feat: combine and pass resource `meta`, URL params and hook's `meta` prop to dataProvider methods (#4135)
refinedev/refine: documentation/docs/api-reference/antd/hooks/field/useSelect/index.md

### `meta`

`meta` is a special property that can be used to pass additional information to data provider methods for the following purposes:

> -   Customize the data provider methods to handle specific use cases.

Reviewer: ```suggestion
-   Customizing the data provider methods for specific use cases.
```

-   Generate GraphQL queries using plain JavaScript Objects (JSON).

> [Refer to the "General Concepts - Meta" documentation for more information &#8594](/docs/api-reference/general-concepts/#meta)

Reviewer: ```suggestion
[Refer to the `meta` section of the General Concepts documentation for more information &#8594](/docs/api-reference/general-concepts/#meta)
```
